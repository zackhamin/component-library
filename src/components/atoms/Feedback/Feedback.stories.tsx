import { Feedback, FeedbackProps } from './Feedback';

export default {
	title: 'atoms/Feedback',
	component: Feedback,
};

export const FeedbackStory = (args: FeedbackProps) => <Feedback {...args} />;

FeedbackStory.storyName = 'Feedback';
FeedbackStory.args = {
	formUrl: 'https://feedback.form',
	children: 'Feedback',
};
