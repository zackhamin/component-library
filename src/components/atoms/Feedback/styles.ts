import { FeedbackProps } from './Feedback';
import styled from 'styled-components';

const StyledFeedback = styled.a<Partial<FeedbackProps>>`
	color: ${({ color }) => color};
	background: ${({ background }) => background};
	display: block;
	font-size: 1.2rem;
	margin: 0;
	padding: 0.4rem 1.6rem;
	position: absolute;
	right: ${({ position }) => (position === 'right' ? '-3.2rem' : 'auto')};
	left: ${({ position }) => (position === 'left' ? '-3.2rem' : 'auto')};
	top: 50vh;
	transform: ${({ position }) =>
		position === 'right' ? 'rotate(-90deg)' : 'rotate(90deg)'};
`;

export { StyledFeedback };
