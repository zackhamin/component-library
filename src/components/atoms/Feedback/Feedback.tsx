import { colors } from 'Styles';
import { ReactNode } from 'react';
import { StyledFeedback } from './styles';

export interface FeedbackProps {
	formUrl: string;
	children: ReactNode;
	color?: string;
	background?: string;
	position?: 'left' | 'right';
}

const Feedback = ({
	formUrl,
	children,
	color = colors.white,
	background = colors.emerald,
	position = 'right',
}: FeedbackProps) => {
	return (
		<StyledFeedback
			href={formUrl}
			target="_blank"
			rel="noopener noreferrer"
			color={color}
			background={background}
			position={position}
		>
			{children}
		</StyledFeedback>
	);
};

export { Feedback };
