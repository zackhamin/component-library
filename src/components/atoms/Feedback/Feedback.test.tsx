import { Feedback } from '../Feedback';
import { render } from '@testing-library/react';

describe('Feedback', () => {
	it('renders correctly', () => {
		const { container } = render(
			<Feedback formUrl="https://feedback.form">Feedback</Feedback>
		);
		expect(container).toMatchSnapshot();
	});
});
