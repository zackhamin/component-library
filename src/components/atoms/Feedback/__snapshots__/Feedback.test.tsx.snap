// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback renders correctly 1`] = `
.c0 {
  color: #ffffff;
  background: #00594C;
  display: block;
  font-size: 1.2rem;
  margin: 0;
  padding: 0.4rem 1.6rem;
  position: absolute;
  right: -3.2rem;
  left: auto;
  top: 50vh;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<div>
  <a
    class="c0"
    color="#ffffff"
    href="https://feedback.form"
    rel="noopener noreferrer"
    target="_blank"
  >
    Feedback
  </a>
</div>
`;
